WooCommerce/Classes/ViewRelated/Dashboard/ProductStock/ProductStockDashboardCardViewModel.swift

```diff
@@ -161,11 +164,10 @@ private extension ProductStockDashboardCardViewModel {
                 }
             }
 
-            while !group.isEmpty {
-                // gather the results and re-throw any failure.
-                if let items = try await group.next() {
-                    allReports.append(contentsOf: items)
-                }
+            // rethrow any failure.
+            for try await items in group {
+                // gather the results
+                allReports.append(contentsOf: items)
```

Comment:
Calling `group.next()` mutates `group`, which will be an error when strict concurrency is enabled. Using `for` loop iterates through the group so it's thread-safe.